### ISSUE TITLE
Annotation probably fails on Windows due to no $EDITOR

### ISSUE THREAD

**Issue by robintw** [recipy/recipy#98]
This code will most probably fail on Windows as there is no $EDITOR (not your PR, the annotate function as it is). 
To make it more robust, on Windows you could use os.startfile to open the file in question with the default application.

(From comment on #97 by @bilderbuchi)

**Comment by cash** [comment]
Just tried this on Windows and recipy annotate does not work because Windows has a different format for environments variables (uses %EDITOR% format).

Changing to:
```python
os.system('%s %s' % (os.environ.get('EDITOR'), f.name))
```
did work for me as long as I set the environment variable first as it is not a standard Windows variable:
```
set EDITOR=notepad.exe
```

**Comment by robintw** [comment]
Thanks. I've done a bit of playing too, and I think something like this will work:

```
editor = os.getenv('EDITOR')
if editor:
    os.system('%s %s' % (editor, f.name))
else:
    touch(f.name)
    os.system(f.name)
```

So, it will try and get the EDITOR variable (which should be set on most Linux/OS X systems) - and if it can't, it will 'touch' the file (like the Unix `touch` command, but in pure Python) and then execute the file which, on Windows, will open the file in Notepad.

For reference, the touch function is defined at http://stackoverflow.com/questions/1158076/implement-touch-using-python

**Comment by cash** [comment]
I assume you're also adding a .txt extension to the temp file to get that last system call to work.

**Comment by robintw** [comment]
Fixed in #155.
